### PR TITLE
resolve halo persistent

### PR DESCRIPTION
**Pull request by newfrenchy83**
When a unit die, this halo is not erased but persist until another unit select.  The weird spacing are accident. Please ignore or correct this and forgive me but github is new for me

### CODE REVIEW

**Comment by ln-zookeeper** [comment]
It's probably obvious, but just to make sure: this should definitely be squash-merged if/when it's merged.

File: src/actions/advancement.cpp [hunk 1]
```diff
@@ -307,6 +307,8 @@ unit_ptr get_amla_unit(const unit &u, const config &mod_option)
 	unit_ptr amla_unit(new unit(u));
 	amla_unit->set_experience(amla_unit->experience_overflow());
 	amla_unit->add_modification("advancement", mod_option);
+	u.anim_comp().clear_haloes() ;
```
Comment (CelticMinstrel) [inline src/actions/advancement.cpp]: Why is there a space before the semicolon?

File: src/scripting/game_lua_kernel.cpp [hunk 2]
```diff
@@ -2105,6 +2105,7 @@ int game_lua_kernel::intf_erase_unit(lua_State *L)
 			if (!map().on_board(loc)) {
 				return luaL_argerror(L, 1, "invalid location");
 			}
+			u->anim_comp(). clear_haloes () ;
```
Comment (CelticMinstrel) [inline src/scripting/game_lua_kernel.cpp]: Please get rid of all the extra spaces in here.

File: src/synced_commands.cpp [hunk 3]
```diff
@@ -433,6 +434,7 @@ SYNCED_COMMAND_HANDLER_FUNCTION(debug_unit, child,  use_undo, /*show*/, /*error_
 		}
 	} else {
 		config cfg;
+		i->anim_comp().clear_haloes () ;
```
Comment (CelticMinstrel) [inline src/synced_commands.cpp]: Please drop all the extra spaces in this line.

File: src/units/drawer.cpp [hunk 4]
```diff
@@ -165,7 +165,9 @@ void unit_drawer::redraw_unit (const unit & u) const
 
 	bool has_halo = ac.unit_halo_ && ac.unit_halo_->valid();
 	if(!has_halo && !u.image_halo().empty()) {
-		ac.unit_halo_ = halo_man.add(x, y - height_adjust, u.image_halo()+u.TC_image_mods(), map_location(-1, -1));
+		ac.unit_halo_ = halo_man.add(0 , 0,u.image_halo()+u.TC_image_mods(), map_location(-1, -1));
```
Comment (CelticMinstrel) [inline src/units/drawer.cpp]: Weird spacing here. I guess not a huge deal, but generally:

* No space before commas
* Space around operators such as `+`

File: src/units/udisplay.cpp [hunk 5]
```diff
@@ -569,6 +569,7 @@ void unit_die(const map_location& loc, unit& loser,
 			unit_animation::hit_type::KILL,secondary_attack,attack,0);
 	animator.start_animations();
 	animator.wait_for_end();
+	loser.anim_comp().clear_haloes ();
```
Comment (CelticMinstrel) [inline src/units/udisplay.cpp]: Please drop the extra space there.

**Comment by jyrkive** [comment]
I see that you have manually reverted commit 0bc068113a9cb8a18f797e4aec29d280653b57d3.

Please explain why. Don't just revert bug fixes for the sake of it!

**Comment by CelticMinstrel** [comment]
Presumably superseded by #1965.

**Pull request closed without merging**